experimenter/tests/integration/nimbus/utils/filter_expression.js

```diff
@@ -1,55 +1,64 @@
+        await TelemetryEnvironment.onInitialized();
+
+        let ASRouterTargeting;
+        try {
+            ASRouterTargeting = ChromeUtils.importESModule("resource:///modules/asrouter/ASRouterTargeting.sys.mjs");
+        } catch (ex) {
+            if (ex.result === Cr.NS_ERROR_FILE_NOT_FOUND) {
+                ASRouterTargeting = ChromeUtils.importESModule("resource://activity-stream/lib/ASRouterTargeting.sys.mjs");
+            } else {
+                throw ex;
+            }
         }
-    }
-    const ExperimentManager = ChromeUtils.importESModule("resource://nimbus/lib/ExperimentManager.sys.mjs");
-    const TargetingContext = ChromeUtils.importESModule("resource://messaging-system/targeting/Targeting.sys.mjs");
-
-    const _experiment = JSON.parse(arguments[1]);
-
-    const context = TargetingContext.TargetingContext.combineContexts(
-        _experiment,
-        {
-            defaultProfile: {},
-            attributionData: {},
-            isMSIX: {},
-            isDefaultHandler: {},
-            defaultPDFHandler: {}
-        }, // Workaround for supporting background tasks
-        ExperimentManager.ExperimentManager.createTargetingContext(),
-        ASRouterTargeting.ASRouterTargeting.Environment
-    );
-    const targetingContext = new TargetingContext.TargetingContext(context);
-    let result = false;
-    try {
-        result = await targetingContext.evalWithDefault(arguments[0]) !== undefined;
+
+        const { ExperimentAPI } = ChromeUtils.importESModule("resource://nimbus/ExperimentAPI.sys.mjs");
+        const TargetingContext = ChromeUtils.importESModule("resource://messaging-system/targeting/Targeting.sys.mjs");
+
+        const _experiment = JSON.parse(recipe);
+        await ExperimentAPI.ready();
+
+        const context = TargetingContext.TargetingContext.combineContexts(
+            _experiment,
+            {
+                defaultProfile: {},
+                attributionData: {},
+                isMSIX: {},
+                isDefaultHandler: {},
+                defaultPDFHandler: {}
+            },
+            ExperimentAPI._manager.createTargetingContext(),
+            ASRouterTargeting.ASRouterTargeting.Environment
+        );
+
+        const targetingContext = new TargetingContext.TargetingContext(context);
+        let result = false;
+        try {
+            result = await targetingContext.evalWithDefault(targetingString) !== undefined;
+        } catch (err) {
+            result = null;
+        }
+
+        callback(result);
     } catch (err) {
-        result = null;
+        callback(null);
     }
-    return result;
 }
 
-let results = remoteSettings(arguments);
+// Get the Selenium callback
+const [targetingString, recipe] = arguments;
+const callback = arguments[arguments.length - 1];
 
-return results;
+remoteSettings(targetingString, recipe, callback);
```

Comment:
Instead of taking the callback as an arg, lets do `remoteSettings(targetingString, recipe).then(callback, () => null)`

It makes the control flow of `remoteSettings` cleaner and it doesnt need a bunch of try/catch.